### PR TITLE
New Resource: `azurerm_data_factory_trigger_schedule` 

### PR DESCRIPTION
**Pull request by mbfrahry**
```
--- PASS: TestAccAzureRMDataFactoryTriggerSchedule_basic (84.33s)
--- PASS: TestAccAzureRMDataFactoryTriggerSchedule_complete (103.15s)
```

### CODE REVIEW

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	"fmt"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: could we run goimports on this file?

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMDataFactoryPipelineName,
+			},
+
+			// There's a bug in the Azure API where this is returned in lower-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/5788
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: I suggest reopening this on the sdk as they seem to actually act on issues there... sometimes 😪

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMDataFactoryPipelineName,
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: validateAzureRMDataFactoryPipelineName? is that the same rules as for a trigger name?

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"data_factory_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid data_factory_name, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules`,
+				),
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: We should probably put this into a shared validation function as its used in multiple places

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+				Default:      1,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+
+			"timezone": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "UTC",
+				ValidateFunc: validate.NoEmptyStrings,
+			},
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: We have a few different timezone validation functions, perhaps one would work here?

File: azurerm/resource_arm_data_factory_trigger_schedule.go [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule.go]: Could we validate against `NoEmptyStrings`?

File: azurerm/resource_arm_data_factory_trigger_schedule_test.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_data_factory_trigger_schedule" "test" {
+  name                = "acctestdf%d"
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule_test.go]: Could we put some capitals into the name?
```suggestion
  name                = "acctestDFTS%d"
```

File: azurerm/resource_arm_data_factory_trigger_schedule_test.go [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMDataFactoryTriggerSchedule_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/resource_arm_data_factory_trigger_schedule_test.go]: I've been doing this lately for tests
```suggestion
  name     = "acctestRG-datafactory-%d"
```

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  interval    = 5
+  frequency   = "Day"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Data Factory Schedule Trigger. Changing this forces a new resource to be created. Must be globally unique. See the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/data-factory/naming-rules) for all restrictions.
```
Comment (katbyte) [inline website/docs/r/data_factory_trigger_schedule.html.markdown]: If the API has a `CheckName1 function we should probably make use of it to ensure the name is unique

Comment (mbfrahry) [reply website/docs/r/data_factory_trigger_schedule.html.markdown]: Unfortunately, I didn't find one

File: azurerm/resource_arm_data_factory_trigger_schedule.go (hunk 3 above)
Comment (mbfrahry) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: Sure is. Moved it out into it's own validation

File: azurerm/resource_arm_data_factory_trigger_schedule.go (hunk 5 above)
Comment (mbfrahry) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: I'm hesitant to add any validation here and might actually just remove this value. It doesn't seem to do anything and the UI just defaults the time to UTC anyway

**Comment by mbfrahry** [comment]
Review addressed but I need to add one more piece before it's ready

File: azurerm/resource_arm_data_factory_trigger_schedule.go (hunk 5 above)
Comment (katbyte) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: If you pick PST the API still takes the value as UTC?

Then i would definitely remove it, add a note that the time is always UC, and open a bug on the SDK with a link in the code explaining why UTC has been ignored.

Comment (mbfrahry) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: Confirmed that timezone doesn't work no matter the value. It's been removed and an issue has been opened https://github.com/Azure/azure-sdk-for-go/issues/6244

Comment (mbfrahry) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: The Data Factory API attributes tend to accept any value without validation. You can submit anything to timezone and it'll be accepted whether it's valid or not but `Pacific Standard Time` didn't change the format of the time at all so I've submitted an issue on the sdk as a start. https://github.com/Azure/azure-sdk-for-go/issues/6244

**Comment by mbfrahry** [comment]
This should be ready for a final review @katbyte

File: azurerm/resource_arm_data_factory_trigger_schedule.go (hunk 5 above)
Comment (katbyte) [reply azurerm/resource_arm_data_factory_trigger_schedule.go]: _fun_

**Comment by neil-yechenwei** [comment]
@mbfrahry , I noticed you made a PR (https://github.com/terraform-providers/terraform-provider-azurerm/pull/3244) to add support for azurerm_data_factory_pipeline. I found you added a property "activity" in your first commit (https://github.com/terraform-providers/terraform-provider-azurerm/pull/3244/commits/06bd3226b5d97c47b4d33a542231d624c1e31b5e). But you deleted this property in later commit (https://github.com/terraform-providers/terraform-provider-azurerm/pull/3244/commits/6822cfa94215b4e0132114e3da0bb192596ad7af). After learned from the azure doc for data factory (https://docs.microsoft.com/en-us/azure/data-factory/v1/data-factory-copy-activity-tutorial-using-powershell), I think you should retain this property "activity" since I think it's an important thing for data factory pipeline to combine the InputDataset and the OutputDataset. May I know why you removed the property "activity"? If this property is necessary, can it be added back?

**Pull request merged**